### PR TITLE
Added github actions for code coverage ensurity

### PR DESCRIPTION
**Pull request by Varunaditya1**
*Issue #, if available:*

### Description of changes:

Added github actions for ensuring code coverage. 

### Test Steps

```
cmake -S test/unit-test -B build/ -G "Unix Makefiles"  -DBUILD_CLONE_SUBMODULES=ON -DCMAKE_C_FLAGS='--coverage -Wall -Wextra -Werror -DNDEBUG -DLIBRARY_LOG_LEVEL=LOG_DEBUG'
make -C build all
cd build
ctest -E system --output-on-failure
make coverage
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+name: CI Checks
+on:
+  push:
+    branches: ["**"]
```
Comment (thomas-roos) [inline .github/workflows/ci.yml:4]: what does this exactly? ["**"]

Comment (thomas-roos) [reply .github/workflows/ci.yml:4]: and why is it necessary?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  pull_request:
+    branches: [main]
+  workflow_dispatch:
+jobs:
+  unittest:
+    runs-on: ubuntu-latest
```
Comment (thomas-roos) [inline .github/workflows/ci.yml:10]: I prefer to specify a version to not have surprises.

**Review by thomas-roos — APPROVED** [review]

**Review by aggarg — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (aggarg) [reply .github/workflows/ci.yml:4]: This mean run this check on a push to any branch. Useful when we develop changes in our fork.

Comment (thomas-roos) [reply .github/workflows/ci.yml:4]: Isn't this the default?

**Review by ActoryOu — APPROVED** [review]

**Pull request merged**